device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -820,21 +901,80 @@ private void handleDisconnection(TransportException transportException)
         reconnect(transportException);
     }
 
+    // should only be called when multiplexing an only a particular device went offline
+    private void reconnectDeviceSession(TransportException transportException, String deviceId)
+    {
+        long reconnectionStartTimeMillis = System.currentTimeMillis();
+        int reconnectionAttempt = 0;
+        boolean hasReconnectOperationTimedOut = this.hasOperationTimedOut(reconnectionStartTimeMillis, deviceId);
+        RetryDecision retryDecision = null;
+
+        while (this.deviceConnectionStates.get(deviceId) == IotHubConnectionStatus.DISCONNECTED_RETRYING
+                && !hasReconnectOperationTimedOut)
+        {
+            this.log.trace("Attempting device session reconnect attempt {}", reconnectionAttempt);
+            reconnectionAttempt++;
+
+            RetryPolicy retryPolicy = this.getConfig(deviceId).getRetryPolicy();
+            retryDecision = retryPolicy.getRetryDecision(reconnectionAttempt, transportException);
+            if (!retryDecision.shouldRetry())
+            {
+                break;
+            }
+
+            // This call triggers some async amqp logic, so all this function can do is wait for a bit and check the connection
+            // status for this device before retrying.
+            singleDeviceReconnectAttemptAsync(deviceId);
+
+            this.log.trace("Sleeping between device reconnect attempts for device {}", deviceId);
+            IotHubTransport.sleepUninterruptibly(retryDecision.getDuration(), MILLISECONDS);
+
+            hasReconnectOperationTimedOut = this.hasOperationTimedOut(reconnectionStartTimeMillis);
+        }
+
+        // reconnection may have failed, so check last retry decision, check for timeout, and check if last exception
+        // was terminal
+        try
+        {
+            if (retryDecision != null && !retryDecision.shouldRetry())
+            {
```

Comment:
I mentioned it elsewhere, but don't see the comment anymore. Bracing here seems to follow .NET style guidelines, not Java.

Reply:
We have wildly different bracing style in track 1 repos than what we have in track 2. This is standard in our track 1 repo so far, albeit we could certainly change

Reply:
Hey, I'm not going to complain. Let's embrace .NET style everywhere. :)